packages/create-project/src/create-project.ts

```diff
@@ -0,0 +1,188 @@
+import retry from 'async-retry';
+import chalk from 'chalk';
+import path from 'path';
+import {
+  downloadAndExtractExample,
+  downloadAndExtractRepo,
+  getRepoInfo,
+  hasExample,
+  hasRepo,
+  RepoInfo,
+} from './helpers/examples';
+import { makeDir } from './helpers/make-dir';
+import { tryGitInit } from './helpers/git';
+import { install } from './helpers/install';
+import { isFolderEmpty } from './helpers/is-folder-empty';
+import { getOnline } from './helpers/is-online';
+// import { shouldUseYarn } from './helpers/should-use-yarn';
+import { isWriteable } from './helpers/is-writeable';
+
+export class DownloadError extends Error {}
+
+export async function createApp({
+  appPath,
+  useYarn,
+  example,
+  examplePath,
+}: {
+  appPath: string;
+  useYarn: boolean;
+  example: string;
+  examplePath?: string;
+}): Promise<void> {
+  let repoInfo: RepoInfo | undefined;
+
+  if (example) {
+    let repoUrl: URL | undefined;
+
+    try {
+      repoUrl = new URL(example);
+    } catch (error: any) {
+      if (error.code !== 'ERR_INVALID_URL') {
+        console.error(error);
+        process.exit(1);
+      }
+    }
+
+    if (repoUrl) {
+      if (repoUrl.origin !== 'https://github.com') {
+        console.error(
+          `Invalid URL: ${chalk.red(
+            `"${example}"`
+          )}. Only GitHub repositories are supported. Please use a GitHub URL and try again.`
+        );
+        process.exit(1);
+      }
+
+      repoInfo = await getRepoInfo(repoUrl, examplePath);
```

Comment:
Instead of returning undefined, you can just let this throw and log in the `catch` clause.